integreat_cms/cms/utils/linkcheck_utils.py

```diff
@@ -41,13 +47,37 @@ def get_urls(
         # If the results should be limited to specific ids, filter the queryset
         urls = urls.filter(id__in=url_ids)
     if region_slug:
+        region = Region.objects.get(slug=region_slug)
+        latest_pagetranslation_versions = list(
+            PageTranslation.objects.filter(
+                page__id__in=Subquery(region.non_archived_pages.values("pk")),
+            )
+            .distinct("page__id", "language__id")
+            .values_list("pk", flat=True)
+        )
+        latest_poitranslation_versions = list(
+            POITranslation.objects.filter(poi__region=region)
+            .distinct("poi__id", "language__id")
+            .values_list("pk", flat=True)
+        )
+        latest_eventtranslation_versions = list(
+            EventTranslation.objects.filter(event__region=region)
+            .distinct("event__id", "language__id")
+            .values_list("pk", flat=True)
+        )
+        latest_imprinttranslation_versions = list(
+            ImprintPageTranslation.objects.filter(page__region=region)
+            .distinct("page__id", "language__id")
+            .values_list("pk", flat=True)
+        )
         # Get all link objects of the requested region
         region_links = Link.objects.filter(
-            Q(page_translation__page__region__slug=region_slug)
-            | Q(imprint_translation__page__region__slug=region_slug)
-            | Q(event_translation__event__region__slug=region_slug)
-            | Q(poi_translation__poi__region__slug=region_slug)
+            Q(page_translation__id__in=latest_pagetranslation_versions)
+            | Q(imprint_translation__id__in=latest_imprinttranslation_versions)
+            | Q(event_translation__id__in=latest_eventtranslation_versions)
+            | Q(poi_translation__id__in=latest_poitranslation_versions)
         ).order_by("id")
+
         if prefetch_content_objects:
             region_links = region_links.prefetch_related("content_object")
```

Comment:
```suggestion
            region_links = region_links.prefetch_related("content_object__language")
```
I noticed that prefetching the language objects also reduces a few queries.
Just the duplicated region queries are hard to eliminate I guess. But that's not different on the develop branch.